sdk/data-generators/shamir-secret-sharing/src/main.rs

```diff
@@ -0,0 +1,258 @@
+//! Data generator sdk/examples/shamir-secret-sharing
```

Comment:
Did you write this yourself? Or did it come from another source?

Reply:
Ah, yes, so this came from a personal side exploration of Galois fields. It fit too well into Veracruz to not put it here and I thought it would be a good example to use on the command line.

Most of the Shamir secret sharing logic came from studying [Wikipedia](https://en.wikipedia.org/wiki/Shamir%27s_Secret_Sharing), [shamir.py](https://github.com/kurtbrose/shamir/blob/master/shamir.py), and some other implementations, the idea of using a separate polynomial for each byte came from Wikipedia I believe, and the GF(8) operations/lookup tables came from an incredibly well written explanation of [Reed-Solomon error correction](https://anadoxin.org/blog/error-recovery-in-raid6.html/).

So this is no way novel and I relied a lot on existing projects, but it was built on principles rather than being copy-pasted from another codebase.

(As an aside, I still don't _really_ understand Galois fields, just enough to provide the basic operations (+,-,*) and to realize how they can satisfy simple algorithms such as Shamir's)

Reply:
That's good enough for me